rustyms/src/glycan/positioned_structure.rs

```diff
@@ -15,14 +15,20 @@ use crate::{
 
 use crate::uom::num_traits::Zero;
 
+/// The index in the branches as stored in the structure
+pub type GlycanBranchIndex = usize;
+/// The index in the branches when the branches are sorted on mass, this is used to properly render the names of the branches for human consumption
+pub type GlycanBranchMassIndex = usize;
+
 /// Rose tree representation of glycan structure
 #[derive(Debug, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
 pub struct PositionedGlycanStructure {
     pub(super) sugar: MonoSaccharide,
     pub(super) branches: Vec<PositionedGlycanStructure>,
     pub(super) inner_depth: usize,
     pub(super) outer_depth: usize,
-    pub(super) branch: Vec<usize>,
+    /// The branches taken to get to this location (from the root) as the index in the branches and the index in the branches when sorted on mass
+    pub(super) branch: Vec<(GlycanBranchIndex, GlycanBranchMassIndex)>,
```

Comment:
Is this to be assumed to be directed towards the "reducing end"?

Reply:
Yes, I added some more comments to hopefully make it somewhat clearer for other readers later.

```suggestion
    /// For a general glycan with a fucose on the first hexnac and a bisection after the core double 
    /// hexnac + hex, this variable will contain an empty list for the root hexnac. For the fucose
    /// this variable will contain `[(0, 1)]` indicating it is the first branch in the structure but 
    /// the second branch if the branches are sorted by mass. For the monosaccharides in the left 
    /// bisection this variable will contain `[(1, 0), (0, 0)]`, indicating that it took the main 
    /// branch (and not the fucose) and that it took the left branch for the second bisection which
    /// is heavier than the right branch.
```